### PR TITLE
Add ACL Interpolate executors

### PR DESCRIPTION
**Pull request by allnes**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/executors/common/ref_interpolate.cpp [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+            }
+        }
+    });
+}
+
+bool ov::intel_cpu::RefInterpolateExecutor::init(const ov::intel_cpu::InterpolateAttrs &interpolateAttrs,
+                                                 const std::vector<MemoryDescPtr> &srcDescs,
+                                                 const std::vector<MemoryDescPtr> &dstDescs,
+                                                 const dnnl::primitive_attr &attr) {
+    return InterpolateExecutor::init(interpolateAttrs, srcDescs, dstDescs, attr);
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/executors/common/ref_interpolate.cpp:378]: Please check that input parameters are supported. Ref Interpolate supports only nspc layouts and FP32 precision. See Reduce ref executor as a reference.

File: src/plugins/intel_cpu/src/nodes/executors/x64/jit_interpolate.cpp [hunk 2]
```diff
@@ -0,0 +1,1703 @@
...
+    jcp.ID = srcDimPad5d[2];
+    jcp.spatial_dim_size = getSpatialDimsNum(srcDescs[0]->getShape().getDims().size());
+    jcp.layout = interpAttrs.layout;
+    if (jcp.layout != InterpolateLayoutType::planar) {
+        if (mayiuse(cpu::x64::avx512_core)) {
+            interpolateKernel.reset(new jit_uni_interpolate_kernel_f32<cpu::x64::avx512_core>(jcp, *attr.get()));
+        } else if (mayiuse(cpu::x64::avx2)) {
+            interpolateKernel.reset(new jit_uni_interpolate_kernel_f32<cpu::x64::avx2>(jcp, *attr.get()));
+        } else if (mayiuse(cpu::x64::sse41)) {
+            interpolateKernel.reset(new jit_uni_interpolate_kernel_f32<cpu::x64::sse41>(jcp, *attr.get()));
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/executors/x64/jit_interpolate.cpp:1648]: Please use this part of code to properly initialize implType. See https://github.com/dmitry-gorokhov/openvino/blob/poc/arm/src/plugins/intel_cpu/src/nodes/executors/x64/jit_mvn.cpp#L1046-L1075 as a reference

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 3]
```diff
@@ -2094,1162 +741,8 @@ void Interpolate::execute(dnnl::stream strm) {
...
-            return 1;
-        case 2:
-        case 4:
-            return 2;
-        case 5:
-            return 3;
-        default:
-            IE_THROW() << "Can't define number spatial";
-    }
+    srcMemPtr->setDataHandle(const_cast<void *>(reinterpret_cast<const void *>(src_data)));
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp]: That's not legal way to reset source memory. 
I would propose the following:
- Executors still should work with pure memory objects w/o any modifications here (like for other nodes)
- Computations above (for case hasPad=true) should be moved into executors as well. You can put in in base InterpolateExecutor class and reuse between all executors calling it inside exec() method.

File: src/plugins/intel_cpu/src/nodes/executors/common/ref_interpolate.cpp (hunk 1 above)
Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/executors/common/ref_interpolate.cpp:378]: if it used nspc and fp32 checks - all tests become is failed

File: src/plugins/intel_cpu/src/nodes/executors/x64/jit_interpolate.cpp (hunk 2 above)
Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/executors/x64/jit_interpolate.cpp:1648]: done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp (hunk 3 above)
Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: corrected

**Pull request closed without merging**